### DISCUSSION TITLE
cloudlog install issue.

### DISCUSSION THREAD

**Discussion by Grounded-Grid** [magicbug/Cloudlog#1196] [Q&A]
I feel bad posting for help but I'm hoping that someone can point me in the right direction.  I'm certainly no expert but,  I've followed the directions in the wiki, making the changes needed for the distro I'm on, but I can't get past the install wizard.

  When I run the cloudlog install wizard, I'm assuming the directory it's asking for is the document root that apache will be serving cloudlog from.  However, when I give it that, it tells me that the directory can not be found.

  Am I right in assuming that I don't have file system permissions set correctly for apache?  Or is it something else?

Thanks!

**Comment by dg9vh** [comment]
Screenshots could help getting an idea about the problem.

Grounded-Grid ***@***.***> schrieb am Fr., 24. Sept. 2021,
06:18:

**Reply by Grounded-Grid** [reply]
Absolutely, specifically what shots do you want?

**Comment by dg9vh** [comment]
Of your path configuration and the error message

Grounded-Grid ***@***.***> schrieb am Fr., 24. Sept. 2021,
07:26:

**Reply by Grounded-Grid** [reply]
![Screenshot_20210924-015516_Firefox](https://user-images.githubusercontent.com/78862580/134625285-809b51f5-811c-48c9-8952-29a8c1406887.jpg)

![Screenshot_20210924-015328_Firefox](https://user-images.githubusercontent.com/78862580/134625305-a4cb6cc3-0edb-4d6b-886f-665557b22552.jpg)

The path in directory (because you can't read it), copy past is "/srv/http/cloudlog.lawsonawesome.com/"

**Comment by dg9vh** [comment]
And this directory is existing in Filesystem with rwx permissions for the
Webserver?

Grounded-Grid ***@***.***> schrieb am Fr., 24. Sept. 2021,
07:59:

**Reply by Grounded-Grid** [reply]
The directory does exist .  I set permissions per the wiki, subbing in the path to that directory, and in my case apache runs as user "http" member of group "http".  After it didn't work I tried granting permission for the entire cloudlog.lawsonawesome.com/ directory recursively.  Though there's probably some reason you shouldn't do that.

I'll try it again now that I've slept and copy/paste what I type.  Just to cover my bases I'll give permission to that whole directory instead of the individual sub directorys the wiki instructs you to.

**Reply by Grounded-Grid** [reply]
So instead of re-granting permissions, I opted to test that user "http" has read/write permissions instead.
I ran...
`sudo -u http touch /srv/http/cloudlog.lawsonawesome.com`
That worked, so next to prove it has access to the whole directory I ran...
`sudo -u http touch /srv/http/cloudlog.lawsonawesome.com/system/index.php`
that ran, so just for fun I added a line to the comments with...
`sudo -u http vim /srv/http/cloudlog.lawsonawesome.com/index.php`

I don't believe it's a permission problem anymore, it must be an apache problem.
Maybe I need to enable some module let apache access the file system?

Thanks for getting back to me so quickly, it looks like I have some weekend reading to do on apache.

**Reply by T012m3n7oR** [reply]
When checking my config.php in my www/html/application/config dir, my directory value is blank.

What happens if you use blank for your value? (I'm assuming that means root for the website, which would be correct for me as I'm not hosting any other websites out of this.)

**Reply by Grounded-Grid** [reply]
If I leave that field blank, clicking install returns a blank page.
I thought that might mean it was installed, but then connecting to the domain, I still get the note about there not being a config file.

Can I do what the wizard does manually?

**Reply by T012m3n7oR** [reply]
It creates two files:
-database.php  - This defines how to connect to the database
-config.php - The config of the entire install.
It places them both into the application/config file folder.

Then I believe it runs the install/assets/install.sql file against the database.

Can you use "find" or your OS's equivalent and find where it created the files when you said it returned a blank page?  That may give you a hint of where you need to set the directory, another option is to exclude the trailing forward slash and see if that changes anything.

**Reply by Grounded-Grid** [reply]
looking for "database.php" and "config.php" using find  yields only the two in install/config that the wizard uses as a template.

Now, I don't know PHP, but any c-like language kind of reads the same to an outsider.  So it would apear from lines 28-38 in "core_class.php" that leaving the directory field blank is not only acceptable, but the expected input to be given.

When I leave that field blank, and am presented with a blank page, the two files it was suposed to generate are never created.  It's like, somthing about my server is keeping the rest of the script from running after it takes my input.

PHP seems to be running on my server as it should, I should also mention that it is PHP7 and not 8.  Like I said though, I don't know or use PHP, so for all I know there is some sort of config issue keeping the install wizard from compleating.  Now that I see what the wizard does, I think I'll try to create those files and edit them manually.

If cloudlog functions normally after a manual setup, then I'm really stumped as to what my problem was.

Still looking forward to trying cloudlog though, seems to check a lot of boxes for me! haha

**Comment by SpectureCarrier** [comment]
maybe you not open php extensions(mysql or other sql databases)

**Reply by SpectureCarrier** [reply]
Problem solved when I opened the mysql extension in php.ini

**Comment by DL3NO** [comment]
I solved it by NOT entering a directory (leaving this empty).
Installs everything except the known timestamp bug of mariadb/mysql. So I stopped cloudlog. No longer compatible with may mariadb which is used by a couple of other customers.
